### PR TITLE
Added new RBAC TC VerfiyRBACforAppUser

### PR DESCRIPTION
**Pull request by sabrarhussaini**
What this PR does / why we need it:
Added new RBAC TC VerfiyRBACforAppUser
https://portworx.testrail.net/index.php?/cases/view/87889

Which issue(s) this PR fixes (optional)
Closes #PA-1521

Special notes for your reviewer:
Vanilla-s3
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3008/

Vanilla-NFS
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3006/

RKE-S3:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/s3/job/px-backup-system-test-rke/336/

### CODE REVIEW

File: tests/backup/backup_rbac_test.go [hunk 1]
```diff
@@ -1255,3 +1255,291 @@ var _ = Describe("{VerifyRBACForAppAdmin}", func() {
...
+					cloudCredUID = uuid.New()
+					backupLocationUID = uuid.New()
+					credName = fmt.Sprintf("cred-%s-%v", provider, RandomString(6))
+					err = CreateCloudCredential(provider, credName, cloudCredUID, orgID, ctxNonAdmin)
+					dash.VerifyFatal(strings.Contains(err.Error(), "PermissionDenied"), true, fmt.Sprintf("Verifying if App-User [%s] doesn't have permission for creating cloud credentials for provider [%s]", appUser, provider))
+				} else {
+					log.InfoD("Provider NFS does not require creating cloud credential")
+				}
+				backupLocationName = fmt.Sprintf("backup-location-%s-%v", provider, RandomString(4))
+				err = CreateBackupLocationWithContext(provider, backupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "", ctxNonAdmin)
```
Comment (kphalgun-px) [inline tests/backup/backup_rbac_test.go:1331]: Is cloudCredUID initiallzed?

File: tests/backup/backup_rbac_test.go [hunk 2]
```diff
@@ -1255,3 +1255,291 @@ var _ = Describe("{VerifyRBACForAppAdmin}", func() {
...
+			err := DeleteRule(ruleName, orgID, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying deletion of rule [%s] for the Px-Admin user", ruleName))
+		}
+		log.InfoD("Switching context to destination cluster for clean up")
+		err = SetDestinationKubeConfig()
+		log.FailOnError(err, "Unable to switch context to destination cluster [%s]", destinationClusterName)
+		DestroyApps(scheduledAppContexts, opts)
+		log.InfoD("Switching back context to Source cluster")
+		err = SetSourceKubeConfig()
+		log.FailOnError(err, "Unable to switch context to source cluster [%s]", SourceClusterName)
```
Comment (kphalgun-px) [inline tests/backup/backup_rbac_test.go:1543]: How about call it in defer?

```
JustAfterEach(func() {
	defer EndPxBackupTorpedoTest(scheduledAppContexts)
	defer func() {
  	    err = SetSourceKubeConfig()
	    log.FailOnError(err, "Unable to switch context to source cluster [%s]", SourceClusterName)
	}()
    .
    .
    .
}
```

File: tests/backup/backup_rbac_test.go [hunk 3]
```diff
@@ -1255,3 +1255,291 @@ var _ = Describe("{VerifyRBACForAppAdmin}", func() {
...
+		CleanupCloudSettingsAndClusters(backupLocationMap, cloudCredName, cloudCredUID, ctx)
+		UserRules, _ := Inst().Backup.GetAllRules(ctx, orgID)
+		for _, ruleName := range UserRules {
+			err := DeleteRule(ruleName, orgID, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying deletion of rule [%s] for the Px-Admin user", ruleName))
+		}
+		log.InfoD("Switching context to destination cluster for clean up")
+		err = SetDestinationKubeConfig()
+		log.FailOnError(err, "Unable to switch context to destination cluster [%s]", destinationClusterName)
+		DestroyApps(scheduledAppContexts, opts)
```
Comment (kphalgun-px) [inline tests/backup/backup_rbac_test.go:1540]: Not necessary I feel.

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: tests/backup/backup_rbac_test.go [hunk 4]
```diff
@@ -1255,3 +1255,291 @@ var _ = Describe("{VerifyRBACForAppAdmin}", func() {
...
+			dash.VerifyFatal(srcClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			srcClusterUid, err = Inst().Backup.GetClusterUID(ctxNonAdmin, orgID, SourceClusterName)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+			log.Infof("Cluster [%s] uid: [%s]", SourceClusterName, srcClusterUid)
+			dstClusterStatus, err := Inst().Backup.GetClusterStatus(orgID, destinationClusterName, ctxNonAdmin)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", destinationClusterName))
+			dash.VerifyFatal(dstClusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", destinationClusterName))
+			dstClusterUid, err := Inst().Backup.GetClusterUID(ctxNonAdmin, orgID, destinationClusterName)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster uid", destinationClusterName))
+			log.Infof("Cluster [%s] uid: [%s]", destinationClusterName, dstClusterUid)
```
Comment (snigdha-px) [inline tests/backup/backup_rbac_test.go:1461]: dstClusterUid is not required. we can skip this

**Review by snigdha-px — APPROVED** [review]
One very minor comment .Rest LGTM

File: tests/backup/backup_rbac_test.go (hunk 1 above)
Comment (shussaini01) [reply tests/backup/backup_rbac_test.go:1331]: I have now changed the block as below;
image

File: tests/backup/backup_rbac_test.go (hunk 2 above)
Comment (shussaini01) [reply tests/backup/backup_rbac_test.go:1543]: Used the defer.

File: tests/backup/backup_rbac_test.go (hunk 3 above)
Comment (shussaini01) [reply tests/backup/backup_rbac_test.go:1540]: This will be used to destroy the apps created over the destination cluster.

**Comment by ak-px** [comment]
@sabrarhussaini - Do have a run on RKE as well.

File: tests/backup/backup_rbac_test.go (hunk 4 above)
Comment (shussaini01) [reply tests/backup/backup_rbac_test.go]: Done. I have removed both dstClusterUid and srcClusterUid. Realised both are not used in backup and restore validations.

File: tests/backup/backup_rbac_test.go (hunk 1 above)
Comment (sabrarhussaini) [reply tests/backup/backup_rbac_test.go:1330]: Modified the code and removed the cloudcredUID from the if block.
image

File: tests/backup/backup_rbac_test.go (hunk 2 above)
Comment (sabrarhussaini) [reply tests/backup/backup_rbac_test.go:1540]: Addressed the change.

File: tests/backup/backup_rbac_test.go (hunk 4 above)
Comment (sabrarhussaini) [reply tests/backup/backup_rbac_test.go]: Addressed the change.

**Comment by sabrarhussaini** [comment]
@ak-px I have added the RKS run details in the PR;
FYI
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/basic-system-test/job/px/job/s3/job/px-backup-system-test-rke/336/console

**Review by ak-px — APPROVED** [review]
LGTM

File: tests/backup/backup_rbac_test.go [hunk 5]
```diff
@@ -1520,7 +1513,11 @@ var _ = Describe("{VerfiyRBACforAppUser}", func() {
 	})
 
 	JustAfterEach(func() {
-		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+		defer func() {
+			err := SetSourceKubeConfig()
+			log.FailOnError(err, "Unable to switch context to source cluster [%s]", SourceClusterName)
+			EndPxBackupTorpedoTest(scheduledAppContexts)
+		}()
```
Comment (kphalgun-px) [inline tests/backup/backup_rbac_test.go:1520]: How about?

```
		defer EndPxBackupTorpedoTest(scheduledAppContexts)
		defer func() {
			err := SetSourceKubeConfig()
			log.FailOnError(err, "Unable to switch context to source cluster [%s]", SourceClusterName)
		}()
```

If switching context failed I think EndPxBackupTorpedoTest will not run.

Comment (sabrarhussaini) [reply tests/backup/backup_rbac_test.go]: Addressed the change

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Pull request merged**